http/src/main/java/io/micronaut/http/bind/binders/QueryValueArgumentBinder.java

```diff
@@ -130,6 +146,49 @@ public BindingResult<T> bind(ArgumentConversionContext<T> context, HttpRequest<?
         return doBind(context, parameters, BindingResult.unsatisfied());
     }
 
+    private BindingResult<T> bindPojo(ArgumentConversionContext<T> context,
+                                      HttpRequest<?> source,
+                                      AnnotationMetadata annotationMetadata,
+                                      ConvertibleMultiValues<String> parameters,
+                                      Argument<T> argument) {
+        Optional<BeanIntrospection<T>> introspectionOpt = BeanIntrospector.SHARED.findIntrospection(argument.getType());
+        if (introspectionOpt.isEmpty()) {
+            return BindingResult.unsatisfied();
+        }
+
+        BeanIntrospection<T> introspection = introspectionOpt.get();
+        T instance;
+        try {
+            instance = introspection.instantiate();
```

Comment:
Why no-args constructor is required? I'think you need more tests with `record` and required args constructor (without no-arg)